bitnami/kafka/templates/kafka-metrics-role.yaml

```diff
@@ -0,0 +1,23 @@
+{{- if .Values.rbac.create -}}
+apiVersion: {{ include "common.capabilities.rbac.apiVersion" . }}
+kind: Role
+metadata:
+  name: {{ template "kafka.metrics.kafka.name" . }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: metrics
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - services
+    verbs:
+      - get
+      - list
+      - watch

```

Comment:
Could you please justify why these permissions are required?

Reply:
Those are the same permissions as previously applied to the kafka exporter by using the same role/service account as kafka. I am not exactly sure what permissions are required here but I am happy to reduce it further if you have better insights on what is actually required

Reply:
These permissions were set for a very specific purpose: loadbalancer/nodeports auto-discovery when enabling [external access](https://github.com/bitnami/charts/tree/master/bitnami/kafka#accessing-kafka-brokers-from-outside-the-cluster). In the `auto-discovery.sh` script you can find some commands that use `kubectl` to access the K8s API and retrieve information about existing services:

- https://github.com/bitnami/charts/blob/master/bitnami/kafka/templates/scripts-configmap.yaml#L45

However, I don't see any reason to add these permissions to the Kafka metrics pods.

Reply:
Yes, it does not make sense to add the to the kafka exporter pods. The question then is do we need a role at all? Mostly the requirement is that the kafka exporter pods have separate service account and not use the same as the kafka pods

Reply:
> do we need a role at all?

I don't think so. I think it's safe to remove the Role/RoleBinding for this use case.